commit ea0d01e929aae815ce83907804d9ddc045d4a764
Author: Tim Gover <tim.gover@raspberrypi.org>
Date:   Fri Jun 22 12:22:14 2018 +0100

    ASoC: bcm: Refactor to use devm to register cards
    
    The .remove function in many of the BCM drivers just called
    snd_soc_unregister_card. Migrate these drivers to use devm to manage
    card registration allowing the .remove functions to be eliminated.
    
    Drivers performing other operations in .remove have not been updated.
    
    Signed-off-by: Tim Gover <tim.gover@raspberrypi.org>

diff --git a/sound/soc/bcm/adau1977-adc.c b/sound/soc/bcm/adau1977-adc.c
index 83225b9858b4..a0c59a95f0be 100644
--- a/sound/soc/bcm/adau1977-adc.c
+++ b/sound/soc/bcm/adau1977-adc.c
@@ -89,18 +89,13 @@ static int snd_adau1977_adc_probe(struct platform_device *pdev)
 	    }
 	}
 
-	ret = snd_soc_register_card(&snd_adau1977_adc);
+	ret = devm_snd_soc_register_card(&pdev->dev, &snd_adau1977_adc);
 	if (ret && ret != -EPROBE_DEFER)
 		dev_err(&pdev->dev, "snd_soc_register_card() failed: %d\n", ret);
 
 	return ret;
 }
 
-static int snd_adau1977_adc_remove(struct platform_device *pdev)
-{
-	return snd_soc_unregister_card(&snd_adau1977_adc);
-}
-
 static const struct of_device_id snd_adau1977_adc_of_match[] = {
 	{ .compatible = "adi,adau1977-adc", },
 	{},
@@ -115,7 +110,6 @@ static struct platform_driver snd_adau1977_adc_driver = {
 		.of_match_table = snd_adau1977_adc_of_match,
         },
         .probe          = snd_adau1977_adc_probe,
-        .remove         = snd_adau1977_adc_remove,
 };
 
 module_platform_driver(snd_adau1977_adc_driver);
diff --git a/sound/soc/bcm/allo-digione.c b/sound/soc/bcm/allo-digione.c
index c78dd0ab34f4..d65b356cae4e 100644
--- a/sound/soc/bcm/allo-digione.c
+++ b/sound/soc/bcm/allo-digione.c
@@ -229,7 +229,7 @@ static int snd_allo_digione_probe(struct platform_device *pdev)
 			dev_err(&pdev->dev, "devm_gpiod_get() failed\n");
 	}
 
-	ret = snd_soc_register_card(&snd_allo_digione);
+	ret = devm_snd_soc_register_card(&pdev->dev, &snd_allo_digione);
 	if (ret && ret != -EPROBE_DEFER)
 		dev_err(&pdev->dev, "snd_soc_register_card() failed: %d\n",
 			ret);
@@ -237,11 +237,6 @@ static int snd_allo_digione_probe(struct platform_device *pdev)
 	return ret;
 }
 
-static int snd_allo_digione_remove(struct platform_device *pdev)
-{
-	return snd_soc_unregister_card(&snd_allo_digione);
-}
-
 static const struct of_device_id snd_allo_digione_of_match[] = {
 	{ .compatible = "allo,allo-digione", },
 	{},
@@ -255,7 +250,6 @@ static struct platform_driver snd_allo_digione_driver = {
 		.of_match_table	= snd_allo_digione_of_match,
 	},
 	.probe  = snd_allo_digione_probe,
-	.remove = snd_allo_digione_remove,
 };
 
 module_platform_driver(snd_allo_digione_driver);
diff --git a/sound/soc/bcm/allo-piano-dac.c b/sound/soc/bcm/allo-piano-dac.c
index cfe81c946271..df5831763aa4 100644
--- a/sound/soc/bcm/allo-piano-dac.c
+++ b/sound/soc/bcm/allo-piano-dac.c
@@ -90,7 +90,7 @@ static int snd_allo_piano_dac_probe(struct platform_device *pdev)
 			pdev->dev.of_node, "allo,24db_digital_gain");
 	}
 
-	ret = snd_soc_register_card(&snd_allo_piano_dac);
+	ret = devm_snd_soc_register_card(&pdev->dev, &snd_allo_piano_dac);
 	if (ret && ret != -EPROBE_DEFER)
 		dev_err(&pdev->dev,
 			"snd_soc_register_card() failed: %d\n", ret);
@@ -98,11 +98,6 @@ static int snd_allo_piano_dac_probe(struct platform_device *pdev)
 	return ret;
 }
 
-static int snd_allo_piano_dac_remove(struct platform_device *pdev)
-{
-	return snd_soc_unregister_card(&snd_allo_piano_dac);
-}
-
 static const struct of_device_id snd_allo_piano_dac_of_match[] = {
 	{ .compatible = "allo,piano-dac", },
 	{ /* sentinel */ },
@@ -116,7 +111,6 @@ static struct platform_driver snd_allo_piano_dac_driver = {
 		.of_match_table = snd_allo_piano_dac_of_match,
 	},
 	.probe          = snd_allo_piano_dac_probe,
-	.remove         = snd_allo_piano_dac_remove,
 };
 
 module_platform_driver(snd_allo_piano_dac_driver);
diff --git a/sound/soc/bcm/audioinjector-octo-soundcard.c b/sound/soc/bcm/audioinjector-octo-soundcard.c
index 2ad83c40473f..6029a12d77cc 100644
--- a/sound/soc/bcm/audioinjector-octo-soundcard.c
+++ b/sound/soc/bcm/audioinjector-octo-soundcard.c
@@ -308,19 +308,12 @@ static int audioinjector_octo_probe(struct platform_device *pdev)
 			}
 	}
 
-	ret = snd_soc_register_card(card);
+	ret = devm_snd_soc_register_card(&pdev->dev, card);
 	if (ret != 0)
 		dev_err(&pdev->dev, "snd_soc_register_card failed (%d)\n", ret);
 	return ret;
 }
 
-static int audioinjector_octo_remove(struct platform_device *pdev)
-{
-	struct snd_soc_card *card = platform_get_drvdata(pdev);
-
-	return snd_soc_unregister_card(card);
-}
-
 static const struct of_device_id audioinjector_octo_of_match[] = {
 	{ .compatible = "ai,audioinjector-octo-soundcard", },
 	{},
@@ -334,7 +327,6 @@ static struct platform_driver audioinjector_octo_driver = {
 		.of_match_table = audioinjector_octo_of_match,
 	},
 	.probe	= audioinjector_octo_probe,
-	.remove	= audioinjector_octo_remove,
 };
 
 module_platform_driver(audioinjector_octo_driver);
diff --git a/sound/soc/bcm/audioinjector-pi-soundcard.c b/sound/soc/bcm/audioinjector-pi-soundcard.c
index 491906bbf446..6fc42570a141 100644
--- a/sound/soc/bcm/audioinjector-pi-soundcard.c
+++ b/sound/soc/bcm/audioinjector-pi-soundcard.c
@@ -156,19 +156,12 @@ static int audioinjector_pi_soundcard_probe(struct platform_device *pdev)
 			}
 	}
 
-	if ((ret = snd_soc_register_card(card))) {
+	if ((ret = devm_snd_soc_register_card(&pdev->dev, card))) {
 		dev_err(&pdev->dev, "snd_soc_register_card failed (%d)\n", ret);
 	}
 	return ret;
 }
 
-static int audioinjector_pi_soundcard_remove(struct platform_device *pdev)
-{
-	struct snd_soc_card *card = platform_get_drvdata(pdev);
-	return snd_soc_unregister_card(card);
-
-}
-
 static const struct of_device_id audioinjector_pi_soundcard_of_match[] = {
 	{ .compatible = "ai,audioinjector-pi-soundcard", },
 	{},
@@ -182,7 +175,6 @@ static struct platform_driver audioinjector_pi_soundcard_driver = {
 		.of_match_table = audioinjector_pi_soundcard_of_match,
        },
        .probe          = audioinjector_pi_soundcard_probe,
-       .remove         = audioinjector_pi_soundcard_remove,
 };
 
 module_platform_driver(audioinjector_pi_soundcard_driver);
diff --git a/sound/soc/bcm/digidac1-soundcard.c b/sound/soc/bcm/digidac1-soundcard.c
index 63204f3e33e1..b99a0c858ebe 100644
--- a/sound/soc/bcm/digidac1-soundcard.c
+++ b/sound/soc/bcm/digidac1-soundcard.c
@@ -386,7 +386,7 @@ static int digidac1_soundcard_probe(struct platform_device *pdev)
 		}
 	}
 
-	ret = snd_soc_register_card(&digidac1_soundcard);
+	ret = devm_snd_soc_register_card(&pdev->dev, &digidac1_soundcard);
 	if (ret && ret != -EPROBE_DEFER)
 		dev_err(&pdev->dev, "snd_soc_register_card() failed: %d\n",
 			ret);
@@ -394,11 +394,6 @@ static int digidac1_soundcard_probe(struct platform_device *pdev)
 	return ret;
 }
 
-static int digidac1_soundcard_remove(struct platform_device *pdev)
-{
-	return snd_soc_unregister_card(&digidac1_soundcard);
-}
-
 static const struct of_device_id digidac1_soundcard_of_match[] = {
 	{ .compatible = "rra,digidac1-soundcard", },
 	{},
@@ -412,7 +407,6 @@ static struct platform_driver digidac1_soundcard_driver = {
 			.of_match_table	= digidac1_soundcard_of_match,
 	},
 	.probe		= digidac1_soundcard_probe,
-	.remove		= digidac1_soundcard_remove,
 };
 
 module_platform_driver(digidac1_soundcard_driver);
diff --git a/sound/soc/bcm/dionaudio_loco-v2.c b/sound/soc/bcm/dionaudio_loco-v2.c
index 8f522deb2a48..192cbfd7baca 100644
--- a/sound/soc/bcm/dionaudio_loco-v2.c
+++ b/sound/soc/bcm/dionaudio_loco-v2.c
@@ -85,7 +85,7 @@ static int snd_rpi_dionaudio_loco_v2_probe(struct platform_device *pdev)
 			pdev->dev.of_node, "dionaudio,24db_digital_gain");
 	}
 
-	ret = snd_soc_register_card(&snd_rpi_dionaudio_loco_v2);
+	ret = devm_snd_soc_register_card(&pdev->dev, &snd_rpi_dionaudio_loco_v2);
 	if (ret)
 		dev_err(&pdev->dev, "snd_soc_register_card() failed: %d\n",
 			ret);
@@ -93,11 +93,6 @@ static int snd_rpi_dionaudio_loco_v2_probe(struct platform_device *pdev)
 	return ret;
 }
 
-static int snd_rpi_dionaudio_loco_v2_remove(struct platform_device *pdev)
-{
-	return snd_soc_unregister_card(&snd_rpi_dionaudio_loco_v2);
-}
-
 static const struct of_device_id dionaudio_of_match[] = {
 	{ .compatible = "dionaudio,dionaudio-loco-v2", },
 	{},
@@ -111,7 +106,6 @@ static struct platform_driver snd_rpi_dionaudio_loco_v2_driver = {
 		.of_match_table = dionaudio_of_match,
 	},
 	.probe          = snd_rpi_dionaudio_loco_v2_probe,
-	.remove         = snd_rpi_dionaudio_loco_v2_remove,
 };
 
 module_platform_driver(snd_rpi_dionaudio_loco_v2_driver);
diff --git a/sound/soc/bcm/dionaudio_loco.c b/sound/soc/bcm/dionaudio_loco.c
index 65e03741d349..5f786436f90e 100644
--- a/sound/soc/bcm/dionaudio_loco.c
+++ b/sound/soc/bcm/dionaudio_loco.c
@@ -85,7 +85,7 @@ static int snd_rpi_dionaudio_loco_probe(struct platform_device *pdev)
 		}
 	}
 
-	ret = snd_soc_register_card(&snd_rpi_dionaudio_loco);
+	ret = devm_snd_soc_register_card(&pdev->dev, &snd_rpi_dionaudio_loco);
 	if (ret && ret != -EPROBE_DEFER)
 		dev_err(&pdev->dev, "snd_soc_register_card() failed: %d\n",
 			ret);
@@ -93,11 +93,6 @@ static int snd_rpi_dionaudio_loco_probe(struct platform_device *pdev)
 	return ret;
 }
 
-static int snd_rpi_dionaudio_loco_remove(struct platform_device *pdev)
-{
-	return snd_soc_unregister_card(&snd_rpi_dionaudio_loco);
-}
-
 static const struct of_device_id snd_rpi_dionaudio_loco_of_match[] = {
 	{ .compatible = "dionaudio,loco-pcm5242-tpa3118", },
 	{ /* sentinel */ },
@@ -111,7 +106,6 @@ static struct platform_driver snd_rpi_dionaudio_loco_driver = {
 		.of_match_table	= snd_rpi_dionaudio_loco_of_match,
 	},
 	.probe  = snd_rpi_dionaudio_loco_probe,
-	.remove = snd_rpi_dionaudio_loco_remove,
 };
 
 module_platform_driver(snd_rpi_dionaudio_loco_driver);
diff --git a/sound/soc/bcm/fe-pi-audio.c b/sound/soc/bcm/fe-pi-audio.c
index aab27c5d02c0..c5686eb15b4b 100644
--- a/sound/soc/bcm/fe-pi-audio.c
+++ b/sound/soc/bcm/fe-pi-audio.c
@@ -123,18 +123,13 @@ static int snd_fe_pi_audio_probe(struct platform_device *pdev)
 	card->dev = &pdev->dev;
 	platform_set_drvdata(pdev, card);
 
-	ret = snd_soc_register_card(card);
+	ret = devm_snd_soc_register_card(&pdev->dev, card);
 	if (ret && ret != -EPROBE_DEFER)
 		dev_err(&pdev->dev, "snd_soc_register_card() failed: %d\n", ret);
 
 	return ret;
 }
 
-static int snd_fe_pi_audio_remove(struct platform_device *pdev)
-{
-	return snd_soc_unregister_card(&fe_pi_audio);
-}
-
 static const struct of_device_id snd_fe_pi_audio_of_match[] = {
 	{ .compatible = "fe-pi,fe-pi-audio", },
 	{},
@@ -148,7 +143,6 @@ static struct platform_driver snd_fe_pi_audio_driver = {
                 .of_match_table = snd_fe_pi_audio_of_match,
         },
         .probe          = snd_fe_pi_audio_probe,
-        .remove         = snd_fe_pi_audio_remove,
 };
 
 module_platform_driver(snd_fe_pi_audio_driver);
diff --git a/sound/soc/bcm/googlevoicehat-soundcard.c b/sound/soc/bcm/googlevoicehat-soundcard.c
index 225854b8e529..314281c7a7d8 100644
--- a/sound/soc/bcm/googlevoicehat-soundcard.c
+++ b/sound/soc/bcm/googlevoicehat-soundcard.c
@@ -89,18 +89,14 @@ static int snd_rpi_googlevoicehat_soundcard_probe(struct platform_device *pdev)
 		}
 	}
 
-	ret = snd_soc_register_card(&snd_rpi_googlevoicehat_soundcard);
+	ret = devm_snd_soc_register_card(&pdev->dev,
+			&snd_rpi_googlevoicehat_soundcard);
 	if (ret)
 		dev_err(&pdev->dev, "snd_soc_register_card() failed: %d\n", ret);
 
 	return ret;
 }
 
-static int snd_rpi_googlevoicehat_soundcard_remove(struct platform_device *pdev)
-{
-	return snd_soc_unregister_card(&snd_rpi_googlevoicehat_soundcard);
-}
-
 static const struct of_device_id snd_rpi_googlevoicehat_soundcard_of_match[] = {
 	{ .compatible = "googlevoicehat,googlevoicehat-soundcard", },
 	{},
@@ -114,7 +110,6 @@ static struct platform_driver snd_rpi_googlevoicehat_soundcard_driver = {
 		.of_match_table = snd_rpi_googlevoicehat_soundcard_of_match,
 	},
 	.probe          = snd_rpi_googlevoicehat_soundcard_probe,
-	.remove         = snd_rpi_googlevoicehat_soundcard_remove,
 };
 
 module_platform_driver(snd_rpi_googlevoicehat_soundcard_driver);
diff --git a/sound/soc/bcm/hifiberry_amp.c b/sound/soc/bcm/hifiberry_amp.c
index 221c6c38e646..7d9d17ced636 100644
--- a/sound/soc/bcm/hifiberry_amp.c
+++ b/sound/soc/bcm/hifiberry_amp.c
@@ -94,7 +94,7 @@ static int snd_rpi_hifiberry_amp_probe(struct platform_device *pdev)
             }
         }
 
-	ret = snd_soc_register_card(&snd_rpi_hifiberry_amp);
+	ret = devm_snd_soc_register_card(&pdev->dev, &snd_rpi_hifiberry_amp);
 
 	if (ret && ret != -EPROBE_DEFER)
 		dev_err(&pdev->dev, "snd_soc_register_card() failed: %d\n", ret);
@@ -103,12 +103,6 @@ static int snd_rpi_hifiberry_amp_probe(struct platform_device *pdev)
 }
 
 
-static int snd_rpi_hifiberry_amp_remove(struct platform_device *pdev)
-{
-	return snd_soc_unregister_card(&snd_rpi_hifiberry_amp);
-}
-
-
 static struct platform_driver snd_rpi_hifiberry_amp_driver = {
         .driver = {
                 .name   = "snd-hifiberry-amp",
@@ -116,7 +110,6 @@ static struct platform_driver snd_rpi_hifiberry_amp_driver = {
 		.of_match_table = snd_rpi_hifiberry_amp_of_match,
         },
         .probe          = snd_rpi_hifiberry_amp_probe,
-        .remove         = snd_rpi_hifiberry_amp_remove,
 };
 
 
diff --git a/sound/soc/bcm/hifiberry_dac.c b/sound/soc/bcm/hifiberry_dac.c
index ee9f13395354..2ef43601f2f9 100644
--- a/sound/soc/bcm/hifiberry_dac.c
+++ b/sound/soc/bcm/hifiberry_dac.c
@@ -89,18 +89,13 @@ static int snd_rpi_hifiberry_dac_probe(struct platform_device *pdev)
 	    }
 	}
 
-	ret = snd_soc_register_card(&snd_rpi_hifiberry_dac);
+	ret = devm_snd_soc_register_card(&pdev->dev, &snd_rpi_hifiberry_dac);
 	if (ret && ret != -EPROBE_DEFER)
 		dev_err(&pdev->dev, "snd_soc_register_card() failed: %d\n", ret);
 
 	return ret;
 }
 
-static int snd_rpi_hifiberry_dac_remove(struct platform_device *pdev)
-{
-	return snd_soc_unregister_card(&snd_rpi_hifiberry_dac);
-}
-
 static const struct of_device_id snd_rpi_hifiberry_dac_of_match[] = {
 	{ .compatible = "hifiberry,hifiberry-dac", },
 	{},
@@ -114,7 +109,6 @@ static struct platform_driver snd_rpi_hifiberry_dac_driver = {
 		.of_match_table = snd_rpi_hifiberry_dac_of_match,
         },
         .probe          = snd_rpi_hifiberry_dac_probe,
-        .remove         = snd_rpi_hifiberry_dac_remove,
 };
 
 module_platform_driver(snd_rpi_hifiberry_dac_driver);
diff --git a/sound/soc/bcm/hifiberry_dacplus.c b/sound/soc/bcm/hifiberry_dacplus.c
index 0946832d5c93..ff61d1ed46c3 100644
--- a/sound/soc/bcm/hifiberry_dacplus.c
+++ b/sound/soc/bcm/hifiberry_dacplus.c
@@ -321,7 +321,8 @@ static int snd_rpi_hifiberry_dacplus_probe(struct platform_device *pdev)
 						"hifiberry-dacplus,slave");
 	}
 
-	ret = snd_soc_register_card(&snd_rpi_hifiberry_dacplus);
+	ret = devm_snd_soc_register_card(&pdev->dev,
+			&snd_rpi_hifiberry_dacplus);
 	if (ret && ret != -EPROBE_DEFER)
 		dev_err(&pdev->dev,
 			"snd_soc_register_card() failed: %d\n", ret);
@@ -329,11 +330,6 @@ static int snd_rpi_hifiberry_dacplus_probe(struct platform_device *pdev)
 	return ret;
 }
 
-static int snd_rpi_hifiberry_dacplus_remove(struct platform_device *pdev)
-{
-	return snd_soc_unregister_card(&snd_rpi_hifiberry_dacplus);
-}
-
 static const struct of_device_id snd_rpi_hifiberry_dacplus_of_match[] = {
 	{ .compatible = "hifiberry,hifiberry-dacplus", },
 	{},
@@ -347,7 +343,6 @@ static struct platform_driver snd_rpi_hifiberry_dacplus_driver = {
 		.of_match_table = snd_rpi_hifiberry_dacplus_of_match,
 	},
 	.probe          = snd_rpi_hifiberry_dacplus_probe,
-	.remove         = snd_rpi_hifiberry_dacplus_remove,
 };
 
 module_platform_driver(snd_rpi_hifiberry_dacplus_driver);
diff --git a/sound/soc/bcm/hifiberry_digi.c b/sound/soc/bcm/hifiberry_digi.c
index 716ac20a8490..8bbd08a61dbe 100644
--- a/sound/soc/bcm/hifiberry_digi.c
+++ b/sound/soc/bcm/hifiberry_digi.c
@@ -241,18 +241,13 @@ static int snd_rpi_hifiberry_digi_probe(struct platform_device *pdev)
 
 	}
 
-	ret = snd_soc_register_card(&snd_rpi_hifiberry_digi);
+	ret = devm_snd_soc_register_card(&pdev->dev, &snd_rpi_hifiberry_digi);
 	if (ret && ret != -EPROBE_DEFER)
 		dev_err(&pdev->dev, "snd_soc_register_card() failed: %d\n", ret);
 
 	return ret;
 }
 
-static int snd_rpi_hifiberry_digi_remove(struct platform_device *pdev)
-{
-	return snd_soc_unregister_card(&snd_rpi_hifiberry_digi);
-}
-
 static const struct of_device_id snd_rpi_hifiberry_digi_of_match[] = {
 	{ .compatible = "hifiberry,hifiberry-digi", },
 	{},
@@ -266,7 +261,6 @@ static struct platform_driver snd_rpi_hifiberry_digi_driver = {
 		.of_match_table = snd_rpi_hifiberry_digi_of_match,
 	},
 	.probe          = snd_rpi_hifiberry_digi_probe,
-	.remove         = snd_rpi_hifiberry_digi_remove,
 };
 
 module_platform_driver(snd_rpi_hifiberry_digi_driver);
diff --git a/sound/soc/bcm/iqaudio_digi.c b/sound/soc/bcm/iqaudio_digi.c
index 8c80e7e2ff24..3d5d85a33654 100644
--- a/sound/soc/bcm/iqaudio_digi.c
+++ b/sound/soc/bcm/iqaudio_digi.c
@@ -203,7 +203,7 @@ static int snd_rpi_iqaudio_digi_probe(struct platform_device *pdev)
 		of_property_read_string(np, prop, &dai->stream_name);
 	}
 
-	ret = snd_soc_register_card(card);
+	ret = devm_snd_soc_register_card(&pdev->dev, card);
 	if (ret && ret != -EPROBE_DEFER)
 		dev_err(&pdev->dev, "snd_soc_register_card() failed: %d\n",
 			ret);
@@ -211,11 +211,6 @@ static int snd_rpi_iqaudio_digi_probe(struct platform_device *pdev)
 	return ret;
 }
 
-static int snd_rpi_iqaudio_digi_remove(struct platform_device *pdev)
-{
-	return snd_soc_unregister_card(&snd_rpi_iqaudio_digi);
-}
-
 static const struct of_device_id snd_rpi_iqaudio_digi_of_match[] = {
 	{ .compatible = "iqaudio,wm8804-digi", },
 	{},
@@ -229,7 +224,6 @@ static struct platform_driver snd_rpi_iqaudio_digi_driver = {
 		.of_match_table	= snd_rpi_iqaudio_digi_of_match,
 	},
 	.probe  = snd_rpi_iqaudio_digi_probe,
-	.remove = snd_rpi_iqaudio_digi_remove,
 };
 
 module_platform_driver(snd_rpi_iqaudio_digi_driver);
diff --git a/sound/soc/bcm/justboom-dac.c b/sound/soc/bcm/justboom-dac.c
index dab0ac7f2f7c..9e7bb9f68ee2 100644
--- a/sound/soc/bcm/justboom-dac.c
+++ b/sound/soc/bcm/justboom-dac.c
@@ -115,7 +115,7 @@ static int snd_rpi_justboom_dac_probe(struct platform_device *pdev)
 			pdev->dev.of_node, "justboom,24db_digital_gain");
 	}
 
-	ret = snd_soc_register_card(&snd_rpi_justboom_dac);
+	ret = devm_snd_soc_register_card(&pdev->dev, &snd_rpi_justboom_dac);
 	if (ret && ret != -EPROBE_DEFER)
 		dev_err(&pdev->dev,
 			"snd_soc_register_card() failed: %d\n", ret);
@@ -123,11 +123,6 @@ static int snd_rpi_justboom_dac_probe(struct platform_device *pdev)
 	return ret;
 }
 
-static int snd_rpi_justboom_dac_remove(struct platform_device *pdev)
-{
-	return snd_soc_unregister_card(&snd_rpi_justboom_dac);
-}
-
 static const struct of_device_id snd_rpi_justboom_dac_of_match[] = {
 	{ .compatible = "justboom,justboom-dac", },
 	{},
@@ -141,7 +136,6 @@ static struct platform_driver snd_rpi_justboom_dac_driver = {
 		.of_match_table = snd_rpi_justboom_dac_of_match,
 	},
 	.probe          = snd_rpi_justboom_dac_probe,
-	.remove         = snd_rpi_justboom_dac_remove,
 };
 
 module_platform_driver(snd_rpi_justboom_dac_driver);
diff --git a/sound/soc/bcm/justboom-digi.c b/sound/soc/bcm/justboom-digi.c
index 88711e5f9194..44527d1f9229 100644
--- a/sound/soc/bcm/justboom-digi.c
+++ b/sound/soc/bcm/justboom-digi.c
@@ -180,7 +180,7 @@ static int snd_rpi_justboom_digi_probe(struct platform_device *pdev)
 	    }
 	}
 
-	ret = snd_soc_register_card(&snd_rpi_justboom_digi);
+	ret = devm_snd_soc_register_card(&pdev->dev, &snd_rpi_justboom_digi);
 	if (ret && ret != -EPROBE_DEFER)
 		dev_err(&pdev->dev,
 			"snd_soc_register_card() failed: %d\n", ret);
@@ -188,11 +188,6 @@ static int snd_rpi_justboom_digi_probe(struct platform_device *pdev)
 	return ret;
 }
 
-static int snd_rpi_justboom_digi_remove(struct platform_device *pdev)
-{
-	return snd_soc_unregister_card(&snd_rpi_justboom_digi);
-}
-
 static const struct of_device_id snd_rpi_justboom_digi_of_match[] = {
 	{ .compatible = "justboom,justboom-digi", },
 	{},
@@ -206,7 +201,6 @@ static struct platform_driver snd_rpi_justboom_digi_driver = {
 		.of_match_table = snd_rpi_justboom_digi_of_match,
 	},
 	.probe          = snd_rpi_justboom_digi_probe,
-	.remove         = snd_rpi_justboom_digi_remove,
 };
 
 module_platform_driver(snd_rpi_justboom_digi_driver);
diff --git a/sound/soc/bcm/rpi-dac.c b/sound/soc/bcm/rpi-dac.c
index 38224467cbab..900889629661 100644
--- a/sound/soc/bcm/rpi-dac.c
+++ b/sound/soc/bcm/rpi-dac.c
@@ -84,18 +84,13 @@ static int snd_rpi_rpi_dac_probe(struct platform_device *pdev)
 		}
 	}
 	
-	ret = snd_soc_register_card(&snd_rpi_rpi_dac);
+	ret = devm_snd_soc_register_card(&pdev->dev, &snd_rpi_rpi_dac);
 	if (ret && ret != -EPROBE_DEFER)
 		dev_err(&pdev->dev, "snd_soc_register_card() failed: %d\n", ret);
 
 	return ret;
 }
 
-static int snd_rpi_rpi_dac_remove(struct platform_device *pdev)
-{
-	return snd_soc_unregister_card(&snd_rpi_rpi_dac);
-}
-
 static const struct of_device_id snd_rpi_rpi_dac_of_match[] = {
 	{ .compatible = "rpi,rpi-dac", },
 	{},
@@ -109,7 +104,6 @@ static struct platform_driver snd_rpi_rpi_dac_driver = {
                 .of_match_table = snd_rpi_rpi_dac_of_match,
         },
         .probe          = snd_rpi_rpi_dac_probe,
-        .remove         = snd_rpi_rpi_dac_remove,
 };
 
 module_platform_driver(snd_rpi_rpi_dac_driver);
diff --git a/sound/soc/bcm/rpi-proto.c b/sound/soc/bcm/rpi-proto.c
index 11f0028ddba6..8227456cfe7d 100644
--- a/sound/soc/bcm/rpi-proto.c
+++ b/sound/soc/bcm/rpi-proto.c
@@ -115,7 +115,7 @@ static int snd_rpi_proto_probe(struct platform_device *pdev)
 		}
 	}
 
-	ret = snd_soc_register_card(&snd_rpi_proto);
+	ret = devm_snd_soc_register_card(&pdev->dev, &snd_rpi_proto);
 	if (ret && ret != -EPROBE_DEFER)
 		dev_err(&pdev->dev,
 				"snd_soc_register_card() failed: %d\n", ret);
@@ -123,12 +123,6 @@ static int snd_rpi_proto_probe(struct platform_device *pdev)
 	return ret;
 }
 
-
-static int snd_rpi_proto_remove(struct platform_device *pdev)
-{
-	return snd_soc_unregister_card(&snd_rpi_proto);
-}
-
 static const struct of_device_id snd_rpi_proto_of_match[] = {
 	{ .compatible = "rpi,rpi-proto", },
 	{},
@@ -142,7 +136,6 @@ static struct platform_driver snd_rpi_proto_driver = {
 		.of_match_table = snd_rpi_proto_of_match,
 	},
 	.probe	  = snd_rpi_proto_probe,
-	.remove	 = snd_rpi_proto_remove,
 };
 
 module_platform_driver(snd_rpi_proto_driver);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1184_ea0d01e929aae815ce83907804d9ddc045d4a764.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1184_ea0d01e929aae815ce83907804d9ddc045d4a764.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
